File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    map<string, int> m;
    
    for(int i= 0; i<participant.size(); i++){
        if(m.find(participant[i]) == m.end()) m[participant[i]]++;
        else m[participant[i]]++;
    }
    
    for(int i = 0; i<completion.size(); i++){
        if(m.find(completion[i]) != m.end()) m[completion[i]]--;
    }
    
    for (auto iter = m.begin() ; iter !=  m.end(); iter++)
    {
        if(iter->second > 0) answer = iter->first;
    }

    return answer;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    let answer = '';
    
    let map = new Map();
    for(let i = 0; i<participant.length; i++){
        if(!map.has(participant[i]))map.set(participant[i], 1);
        else map.set(participant[i], map.get(participant[i]) + 1);
    }
    
    for(let i =0; i<completion.length; i++){
        map.set(completion[i], map.get(completion[i])-1);
    }
    
    for(let[key, value] of map){
        if(value == 1) answer = key;
    }
    
    return answer;
}